src/main/java/com/redhat/labs/lodestar/service/ArtifactService.java

```diff
@@ -0,0 +1,271 @@
+package com.redhat.labs.lodestar.service;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.javers.core.ChangesByObject;
+import org.javers.core.Javers;
+import org.javers.core.JaversBuilder;
+import org.javers.core.diff.Diff;
+import org.javers.core.diff.ListCompareAlgorithm;
+
+import com.redhat.labs.lodestar.model.Artifact;
+import com.redhat.labs.lodestar.model.ArtifactCount;
+import com.redhat.labs.lodestar.model.GetListOptions;
+import com.redhat.labs.lodestar.model.GetOptions;
+import com.redhat.labs.lodestar.model.gitlab.Project;
+
+@ApplicationScoped
+public class ArtifactService {
+
+	@Inject
+	GitService gitService;
+
+	private static final Javers JAVERS = JaversBuilder.javers()
+			.withListCompareAlgorithm(ListCompareAlgorithm.LEVENSHTEIN_DISTANCE).build();
+
+	/**
+	 * Remove all {@link Artifact}s from the database.
+	 */
+	public void purge() {
+		Artifact.removeAllArtifacts();
+	}
+
+	/**
+	 * Fetches all {@link Artifact}s from all projects in the configured Git group
+	 * and inserts into the database.
+	 */
+	public void refresh() {
+
+		// get all projects
+		List<Project> projects = gitService.getProjectsByGroup(true);
+
+		// get artifacts for each project and persist
+		projects.parallelStream().map(gitService::createProjectTree).map(gitService::getArtifactsFile)
+				.map(gitService::parseFile).flatMap(Collection::stream).forEach(a -> {
+
+					// set uuid if missing
+					if (null == a.getUuid()) {
+						a.setUuid(UUID.randomUUID().toString());
+					}
+
+					// persist the artifact
+					createOrUpdateArtifact(a);
+
+				});
+
+	}
+
+	/**
+	 * Processes the given {@link List} of {@link Artifact}s. The artifacts will be
+	 * split based on engagement uuid. Each {@link List} of engagement
+	 * {@link Artifact}s will be updated in both the database and Git.
+	 * 
+	 * @param artifacts
+	 * @param authorEmail
+	 * @param authorName
+	 */
+	public void process(List<Artifact> artifacts, Optional<String> authorEmail, Optional<String> authorName) {
+
+		// split by engagement uuid and process each resulting artifact list
+		processArtifacts(artifacts).entrySet().parallelStream().forEach(e -> {
+
+			// modify artifacts in db
+			String changeLog = modifyArtifactsByEngagementUuid(e.getKey(), e.getValue());
+			// create commit message
+			String commitMessage = new StringBuilder("Changed Engagement Artifacts\n").append(changeLog).toString();
+
+			// pull latest artifacts from db
+			List<Artifact> latest = Artifact.findAllByEngagementUuid(e.getKey());
+			// send to git to update file
+			gitService.createOrUpdateArtifactsFile(e.getKey(), latest, authorEmail, authorName,
+					Optional.ofNullable(commitMessage));
+
+		});
+
+	}
```

Comment:
Is there a scenario where the FE will update artifacts with different engagement uuids? Wondering if we can assume and do more of a `update(artifacts, uuid, commit...)`